### PR TITLE
feat: add `add_to_qflist`, `add_selected`, and `smart_add`

### LINKED ISSUE #634: Quickfix marked files
Hello. How can I add only marked files to quickfix list?
At this moment if i press `ctrl-q` all files from list added to quickfix list despite of marked files.

### PR DESCRIPTION
**Pull request by elianiva**
close #634 

I didn't bind them because I've no idea what is the *good* default mapping, I personally use `<C-q>` and `<A-q>` for `smart_send` and `smart_add` respectively.

cc @Conni2461 @timsofteng

### CODE REVIEW

File: lua/telescope/actions/init.lua [hunk 1]
```diff
@@ -408,18 +408,50 @@ actions.send_to_qflist = function(prompt_bufnr)
...
+
+local smart_send = function(prompt_bufnr, mode)
   local picker = action_state.get_current_picker(prompt_bufnr)
   if table.getn(picker:get_multi_selection()) > 0 then
-    actions.send_selected_to_qflist(prompt_bufnr)
+    if mode == 'r' then
+      actions.send_selected_to_qflist(prompt_bufnr)
+    else
+      actions.add_selected_to_qflist(prompt_bufnr)
+    end
```
Comment (Conni2461) [inline lua/telescope/actions/init.lua:437]: Can't we just use the other local function here and pass in `mode`

Comment (elianiva) [reply lua/telescope/actions/init.lua:437]: oh yeah, we can. Don't know why I didn't thought of that.

**Review by Conni2461 — APPROVED** [review]
LGTM thanks :)

Also you don't have to things if you don't want to do them :laughing: I just thought about this. I don't want to force you into doing a feature you don't want to do :laughing: 

I really really really appreciate your work :)

**Comment by elianiva** [comment]
>  don't want to force you into doing a feature you don't want to do

nah, not at all 😆 
this was really fun to do

**Comment by tami5** [comment]
:heart_eyes:

**Pull request merged**